lib/Conversion/SeqToSV/FirRegLowering.cpp

```diff
@@ -107,53 +115,273 @@ void FirRegLowering::addToIfBlock(OpBuilder &builder, Value cond,
   }
 }
 
+/// Attach an inner-sym to field-id 0 of the given register, or use an existing
+/// inner-sym, if present.
+static StringAttr getInnerSymFor(InnerSymbolNamespace &innerSymNS,
+                                 seq::FirRegOp reg) {
+  auto attr = reg.getInnerSymAttr();
+
+  // If we have an inner sym attribute already, and if there exists a symbol for
+  // field-id 0, then just return that.
+  if (attr)
+    if (auto sym = attr.getSymIfExists(0))
+      return sym;
+
+  // Otherwise, we have to create a new inner sym.
+  auto *context = reg->getContext();
+
+  auto hint = reg.getName();
+
+  // Create our new property for field 0.
+  auto sym = StringAttr::get(context, innerSymNS.newName(hint));
+  auto visibility = StringAttr::get(context, "private");
+  auto property = hw::InnerSymPropertiesAttr::get(context, sym, 0, visibility);
+
+  // Build the new list of inner sym properties.
+  SmallVector<hw::InnerSymPropertiesAttr> properties;
+  if (attr)
+    llvm::append_range(properties, attr.getProps());
```

Comment:
Since the new inner symbol is on lowest fieldID anyway, maybe can just append existing props after it and be done?

If there's any doubt, this is fine too 👍 .